script.js

```diff
@@ -1,12 +1,194 @@
+  navigationContainer.replaceChildren([]);
+
+  let button = document.createElement("button");
+  navigationContainer.appendChild(button);
+  button.id = "back-button";
+  button.className = "back-button";
+  button.innerText = "Back to Library";
+
+  button.addEventListener("click", () => {
+    header.replaceChildren([]);
+    setup();
+  });
+};
+
+// Renders all episode elements stored in `allEpisodes`.
+let renderEpisodeCards = (episodeList) => {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
-}
+  rootElem.replaceChildren([]);
+
+  let episodeListContainer = document.createElement("div");
+  rootElem.appendChild(episodeListContainer);
+  episodeListContainer.className = "episode-list-container";
+
+  // Iterate through `episodeList` and render episode cards and add event listeners for TVMaze episode info.
+  episodeList.forEach((episode) => {
+    let card = document.createElement("article");
+    episodeListContainer.appendChild(card);
+    card.className = "episode-card";
+
+    card.addEventListener("click", () => {
+      parent.open(episode.url);
+    });
+
+    let cardTitle = document.createElement("h2");
+    card.appendChild(cardTitle);
+    cardTitle.innerText = episode.name;
+
+    let combinedEpisodeSeason = document.createElement("div");
+    card.appendChild(combinedEpisodeSeason);
+    let episodeCode = getEpisodeCode(episode);
+    combinedEpisodeSeason.innerText = episodeCode;
+
+    let img = document.createElement("img");
+    card.appendChild(img);
+    img.src = episode.image.medium;
+    img.alt = `${episode.name} front cover`;
+    img.title = episode.name;
+
+    let summary = document.createElement("div");
+    card.appendChild(summary);
+    summary.innerHTML = episode.summary;
+  });
+};
+
+// Render footer with TVMaze licensing agreement.
+let renderFooter = () => {
+  const rootElem = document.getElementById("root");
+  let footer = document.createElement("footer");
+  rootElem.appendChild(footer);
+  footer.className = "footer";
+  footer.innerHTML =
+    "<ul><li>All episode data from: <a href='https://www.tvmaze.com/api#licensing' target='Blank'>TVMaze.com</a></li></ul>";
+};
+
+// Format and return an episode code, format `S01E01`.
+let getEpisodeCode = (episode) => {
+  let code = "";
+  if (episode.season <= 10) {
+    code += `S${episode.season.toString().padStart(2, "0")}`;
+  }
+  if (episode.number <= 10) {
+    code += `E${episode.number.toString().padStart(2, "0")}`;
+  }
+  return code;
+};
+
+// Search `allEpisodes` using the input vale at `#searchBox`, case insensitive,
+// calls render function and updates number of filtered episodes.
+let searchAllEpisodes = () => {
+  let searchEpisodes = document.querySelector("#search-box");
+  const allEpisodes = getAllEpisodes();
+  let filteredEpisodes = allEpisodes.filter((episode) => {
+    return episode.name
+      .toUpperCase()
```

Comment:
converting to uppercase to make it case insensitive is a good strategy. We do it a lot at my work